pkg/maps/eppolicymap/eppolicymap.go

```diff
@@ -79,10 +72,29 @@ func CreateEPPolicyMap() {
 			return
 		}
 
+		EpPolicyMap = bpf.NewMap(mapName,
+			bpf.MapTypeHashOfMaps,
+			&EndpointKey{},
+			int(unsafe.Sizeof(EndpointKey{})),
+			&EPPolicyValue{},
+			int(unsafe.Sizeof(EPPolicyValue{})),
+			MaxEntries,
+			0,
+			0,
+			bpf.ConvertKeyValue,
+		).WithCache()
 		EpPolicyMap.InnerID = uint32(fd)
 	})
 
-	if _, err := EpPolicyMap.OpenOrCreate(); err != nil {
+	_, err := EpPolicyMap.OpenOrCreate()
+	return err
+}
+
+// CreateEPPolicyMap will create both the innerMap (needed for map in map types) and
+// then after BPFFS is mounted create the epPolicyMap. We only create the innerFd once
+// to avoid having multiple inner maps.
+func CreateEPPolicyMap() {
```

Comment:
Maybe we should execute this in an `init()` function?

Reply:
Map creation during init causes problems with our BPFFS path setup doesn't it?

Anyway, both of these maps are created depending on kernel version and Cilium settings. On older kernels we can't (or don't need to) create either of them.